File: go/sample.go
package main

import (
	"fmt"
)

// TransactionType constants
const (
	Income  = "income"
	Expense = "expense"
)

// Transaction struct
type Transaction struct {
	TransactionID   int
	Amount          float64
	TransactionDate string
	TransactionType string
}

// calculateNetProfit calculates the net profit from a slice of Transactions and taxes
func calculateNetProfit(transactions []Transaction, taxes float64) float64 {
	totalIncome := 0.0
	totalExpenses := 0.0

	for _, transaction := range transactions {
		switch transaction.TransactionType {
		case Income:
			totalIncome += transaction.Amount
		case Expense:
			totalExpenses += transaction.Amount
		}
	}

	netIncome := totalIncome - totalExpenses
	netProfit := netIncome - taxes
	return netProfit
}

func main() {
	// Sample transactions
	transactions := []Transaction{
		{TransactionID: 1, Amount: 1000.0, TransactionDate: "2023-01-01", TransactionType: Income},
		{TransactionID: 2, Amount: 500.0, TransactionDate: "2023-01-02", TransactionType: Expense},
		{TransactionID: 3, Amount: 1500.0, TransactionDate: "2023-01-03", TransactionType: Income},
		{TransactionID: 4, Amount: 700.0, TransactionDate: "2023-01-04", TransactionType: Expense},
	}

	taxes := 200.0
	netProfit := calculateNetProfit(transactions, taxes)
	fmt.Printf("Net Profit: %.2f\n", netProfit)
}

File: c/sample.c
#include <stdio.h>
#include <string.h>

typedef struct {
    int transactionId;
    double amount;
    char transactionDate[11]; // Assumes date format is YYYY-MM-DD
    char transactionType[8]; // "income" or "expense"
} Transaction;

double calculateNetProfit(Transaction transactions[], int numTransactions, double taxes) {
    double totalIncome = 0.0;
    double totalExpenses = 0.0;

    for (int i = 0; i < numTransactions; i++) {
        if (strcmp(transactions[i].transactionType, "income") == 0) {
            totalIncome += transactions[i].amount;
        } else if (strcmp(transactions[i].transactionType, "expense") == 0) {
            totalExpenses += transactions[i].amount;
        }
    }

    double netIncome = totalIncome - totalExpenses;
    double netProfit = netIncome - taxes;
    return netProfit;
}

int main() {
    Transaction transactions[4] = {
        {1, 1000.0, "2023-01-01", "income"},
        {2, 500.0, "2023-01-02", "expense"},
        {3, 1500.0, "2023-01-03", "income"},
        {4, 700.0, "2023-01-04", "expense"}
    };

    double taxes = 200.0;
    double netProfit = calculateNetProfit(transactions, 4, taxes);
    printf("Net Profit: %.2f\n", netProfit);

    return 0;
}
